clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -398,9 +398,10 @@ private void updateGroupSubscription(Set<String> topics) {
 
         Set<String> allSubscribedTopics = new HashSet<>();
         Map<String, Subscription> subscriptions = new HashMap<>();
-        for (JoinGroupResponseData.JoinGroupResponseMember memberSubScription : allSubscriptions) {
-            Subscription subscription = ConsumerProtocol.deserializeSubscription(ByteBuffer.wrap(memberSubScription.metadata()));
-            subscriptions.put(memberSubScription.memberId(), subscription);
+        for (JoinGroupResponseData.JoinGroupResponseMember memberSubscription : allSubscriptions) {
```

Comment:
Meta: I'd suggest we do not include the optional instance.id in the constructor, since that is primarily used for deserialization whereas the instance.id is `set` directly; we could instead just add a `setter` of instance.id which is initialized as Null.

In the setter we can also check that instance.id is indeed Null to avoid setting more than once. And by doing it the unit test below can be simplified as well.